modules/express/src/expressApp.ts

```diff
@@ -190,6 +219,30 @@ function checkPreconditions(config: Config) {
     console.warn(`customRootUri or customBitcoinNetwork is set, but env is '${env}'. Setting env to 'custom'.`);
     config.env = 'custom';
   }
+
+  if (env !== 'test' && (externalSignerUrl !== undefined || signerMode !== undefined)) {
+    throw new ExternalSignerConfigError('external signer feature is only enabled for test mode.');
+  }
+
+  if (externalSignerUrl !== undefined && (signerMode !== undefined || signerFileSystemPath !== undefined)) {
+    throw new ExternalSignerConfigError(
+      'signerMode or signerFileSystemPath is set, but externalSignerUrl is also set.'
+    );
+  }
+
+  if ((signerMode !== undefined || signerFileSystemPath !== undefined) && !(signerMode && signerFileSystemPath)) {
+    throw new ExternalSignerConfigError(
+      'signerMode and signerFileSystemPath must both be set in order to run in external signing mode.'
+    );
+  }
+
+  if (signerFileSystemPath !== undefined) {
+    try {
+      checkSignerPrvPath(signerFileSystemPath);
+    } catch (e) {
+      throw e;
+    }
```

Comment:
No need to catch if you're just going to immediately rethrow - feel free to handle in a follow up PR to clean up some other nits as well.
```suggestion
    checkSignerPrvPath(signerFileSystemPath);
```